### PR TITLE
ASoC: SOF: probes support related cleanups

### PR DESCRIPTION
**Pull request by ujfalusi**
Hi,

as requested by @plbossart, I have split out the cleanup patches regarding to the probes support from #3007
The probes conversion to SOF client depends on this series.

### CODE REVIEW

File: sound/soc/sof/pcm.c [hunk 1]
```diff
@@ -899,11 +899,7 @@ void snd_sof_new_platform_drv(struct snd_sof_dev *sdev)
 	pd->pointer = sof_pcm_pointer;
 	pd->ack = sof_pcm_ack;
```
Comment (kv2019i) [inline sound/soc/sof/pcm.c:901]: Maybe add a fixes line pointing to "ASoC: SOF: remove compressed streams". At least I'm left wondering how such code could have ended up in the kernel.... but the above commit explains.

**Review by kv2019i — CHANGES_REQUESTED** [review]
One minor note on a git commit message, otherwise looks great!

File: sound/soc/sof/pcm.c (hunk 1 above)
Comment (ujfalusi) [reply sound/soc/sof/pcm.c:901]: @kv2019i, thanks for pointing it out, I have not looked at the history, but this explains why it was there.
Updated the PR with the Fixes tag (btw: baa58ecfe32950dc1ee98455de5bceafd8520463 is not "upstream")

Comment (ujfalusi) [reply sound/soc/sof/pcm.c:901]: I stand corrected, the patch should apply upstream as well, we have the same remains there, but it is a bit problematic to find the commit to blame there.

Comment (plbossart) [reply sound/soc/sof/pcm.c:901]: or remove the Fixes tag, it's not really that bad. This removes dead-code, that's more of an improvement.

**Review by plbossart — COMMENTED** [review]
looks ok @ujfalusi only the second commit message needs to be updated. I suggest removing the fixes tag.

**Comment by ujfalusi** [comment]
@plbossart, thank you, I'll update the PR in few minutes to remove the Fixes tag.

**Comment by ujfalusi** [comment]
Changes since v2:
- removed fixes tag from patch two

Changes since v1:
- added fixes tag to commit two

**Review by plbossart — DISMISSED** [review]
LGTM @ujfalusi thanks for the cleanup!

**Review by dbaluta — DISMISSED** [review]

**Comment by dbaluta** [comment]
Looks good to me! Thanks @ujfalusi.

**Comment by ujfalusi** [comment]
Hi,

pushed one more patch to use correct postfix for the functions used in snd_soc_cdai_ops for startup and shutdown.

Sorry for the late update but I wanted to include it to this PR and not take it via the auxbus support.

**Review by plbossart — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Pull request merged**